package.json

```diff
@@ -39,6 +39,7 @@
     ],
     "babel": "inherit"
   },
+  "types": "src/index.d.ts",
```

Comment:
Are you sure this is right? I tested this branch and needed to change it to `./src/index.d.ts` to work.

Reply:
that seems to be the better path per http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

I will update that separate from this PR since I am still not clear how we want to move forward with the interface type + type formatting